packages/data-stores/src/domain-suggestions/queries.ts

```diff
@@ -0,0 +1,67 @@
+import { useQuery } from '@tanstack/react-query';
+import { stringify } from 'qs';
+import wpcomProxyRequest from 'wpcom-proxy-request';
+import { getFormattedPrice, normalizeDomainSuggestionQuery } from './utils';
+import type { DomainSuggestion, DomainSuggestionSelectorOptions } from './types';
+
+export function getDomainSuggestionsQueryKey(
+	search: string,
+	options: DomainSuggestionSelectorOptions = {}
+) {
+	return [ 'domain-suggestions', search, options ];
+}
+
+export function useGetDomainSuggestions(
+	search: string,
+	searchOptions: DomainSuggestionSelectorOptions = {},
+	queryOptions = {}
+) {
+	return useQuery( {
+		queryKey: getDomainSuggestionsQueryKey( search, searchOptions ),
+		queryFn: async () => {
+			const queryObject = normalizeDomainSuggestionQuery( search, searchOptions );
+			if ( ! queryObject.query ) {
+				throw new Error( 'Empty query' );
+			}
+			const suggestions: DomainSuggestion[] = await wpcomProxyRequest( {
+				apiVersion: '1.1',
+				path: '/domains/suggestions',
+				query: stringify( queryObject ),
+			} );
+			if ( ! Array.isArray( suggestions ) ) {
+				throw new Error( 'Invalid response from the server' );
+			}
+
+			const processedSuggestions = suggestions.map( ( suggestion: DomainSuggestion ) => {
+				if ( suggestion.unavailable ) {
+					return suggestion;
+				}
+				return {
+					...suggestion,
+					...( suggestion.raw_price &&
+						suggestion.currency_code && {
+							cost: getFormattedPrice( suggestion.raw_price, suggestion.currency_code ),
+						} ),
+				};
+			} );
+
+			return processedSuggestions;
+		},
+		enabled: !! search,
+		staleTime: 1000 * 60 * 5, // 5 minutes
```

Comment:
Maybe we can pull this out into a constant. 

Also a little concerning maybe that there's no consistency or design across most queries in data stores concerning this expiration time (some set to Infinity, others to some value). could this be configured better?

Reply:
> Maybe we can pull this out into a constant.

Done via 8e0788fce420e5ffc8a721c749804295c2ab1c00.

> Also a little concerning maybe that there's no consistency or design across most queries in data stores concerning this expiration time (some set to Infinity, others to some value). could this be configured better?

I think it depends on the query too. In some cases, the server data will never change, so it is safe to set the staleTime to Infinity. In our case, there is a small chance that the free subdomain could be picked up by some other user, so we set a finite expiration.